### PR TITLE
Add GCI tests for kubernetes-e2e-gke-1-4, kubernetes-e2e-gke-1-3

### PR DESCRIPTION
**Pull request by mtaufen**
Job,GCI Project
`kubernetes-e2e-gci-gke-release-1.4`,`k8s-jkns-gci-gke-1-4`
`kubernetes-e2e-gci-gke-serial-release-1.4`,`k8s-jkns-gci-gke-serial-1-4`
`kubernetes-e2e-gci-gke-slow-release-1.4`,`k8s-jkns-gci-gke-slow-1-4`
`kubernetes-e2e-gci-gke-reboot-release-1.4`,`k8s-jkns-gci-gke-reboot-1-4`
`kubernetes-e2e-gci-gke-ingress-release-1.4`,`k8s-jkns-gci-gke-ingress-1-4`
`kubernetes-e2e-gci-gke-release-1.3`,`k8s-jkns-gci-gke-1-3`
`kubernetes-e2e-gci-gke-serial-release-1.3`,`k8s-jkns-gci-gke-serial-1-3`
`kubernetes-e2e-gci-gke-slow-release-1.3`,`k8s-jkns-gci-gke-slow-1-3`
`kubernetes-e2e-gci-gke-reboot-release-1.3`,`k8s-jkns-gci-gke-reboot-1-3`
`kubernetes-e2e-gci-gke-ingress-release-1.3`,`kubernetes-gci-gke-ingress-1-3`

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/585)

### CODE REVIEW

**Comment by maisem** [comment]
LGTM - deferring to @spxtr to take a look as well.

**Comment by spxtr** [comment]
Do we really need 1.2 jobs?

**Comment by vishh** [comment]
@maisem should we ignore release-1.2?

**Comment by mtaufen** [comment]
Removed 1.2 jobs for the time being.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml [hunk 1]
```diff
@@ -303,6 +357,71 @@
...
+                export KUBE_GKE_IMAGE_TYPE="gci"
+        - 'gci-gke-reboot-release-1.3':  # kubernetes-e2e-gci-gke-reboot-release-1.3
+            description: 'Run [Feature:Reboot] tests on GKE on the release-1.3 branch, using GCI image.'
+            timeout: 180
+            job-env: |
+                export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Reboot\]"
+                export JENKINS_PUBLISHED_VERSION="ci/latest-1.3"
+                export PROJECT="k8s-jkns-gci-gke-reboot-1-3"
+                export KUBE_GKE_IMAGE_TYPE="gci"
+        - 'gci-gke-ingress-release-1.2':  # kubernetes-e2e-gci-gke-ingress-release-1.2
```
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml]: eek!

**Comment by spxtr** [comment]
Other than the eek! looks good.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml (hunk 1 above)
Comment (mtaufen) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml]: Then I'm assuming the `gke-ingress-release-1.2` suffix shouldn't be in `kubernetes-e2e-gke-1-3` tests either?

Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml]: Agreed.

Comment (mtaufen) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml]: Ok done.

**Pull request merged**